### ISSUE TITLE
step 3 fails with "IndexError(list index out of range)"

### ISSUE THREAD

**Issue by reutha** [dereneaton/ipyrad#215]
ipyrad version 0.5.15, pairgbs mode.
Step 6 fails with the following:
Step 6: Clustering at 0.95 similarity across 72 samples
  [####################] 100%  concat/shuffle input  | 0:01:42
  [####################] 100%  clustering across     | 9:48:03
  [####################] 100%  building clusters     | 0:01:10
  [####################] 100%  aligning clusters     | 1:34:25
  Encountered an error, see ./ipyrad_log.txt.
  ('error in step 6 %s', <Remote[15]:IndexError(list index out of range)>)

unfortunately, ipyrad_log.txt was deleted, I am trying to reproduce on a smaller sample set.

**Comment by isaacovercast** [comment]
Any update on status? Did you manage to get this working with a newer version of ipyrad?

**Comment by reutha** [comment]
Haven't tried a new version yet.

**Comment by isaacovercast** [comment]
We haven't seen this bug in a long time, so i'm assuming it was fixed as part of one of the newer releases. This was hitting 0.5.15 which is ancient.

**Comment by angelajmcd** [comment]
I'm using ipyrad v.0.7.15 and am getting this error after updating from 0.7.13.

The log file says ERROR IndexError(list index out of range).  I've checked my barcodes and sample names to make sure they match up. I also thought maybe it was a threading issue, so I've tried threading jobs a few different ways but I still get the error. 

Any ideas?

**Comment by isaacovercast** [comment]
Hi Angela,
Can you post the last few relevant lines of the ipyrad_log.txt? Also, if
you can post the progress of the step so i can get an idea where it's
failing that'll help. Had you run the previous steps using v0.7.13 and then
upgraded to 0.7.15? Is this a denovo or reference based alignment?

On Wed, Oct 11, 2017 at 11:11 AM, Angela McDonnell <notifications@github.com

**Comment by angelajmcd** [comment]
Isaac,

Here is the ipyrad_log.txt:
  Begin run: 2017-10-10 09:42
  Using args {'preview': False, 'force': False, 'threads': 2, 'results': False, 'quiet': False, 'merge': None, 'ipcluster': None, 'cores': 0, 'params': 'params-allsab1.txt', 'branch': None, 'steps': '1234567', 'debug': False, 'new': None, 'MPI': False}
  Platform info: ('Darwin', 'biol327-mac', '13.4.0', 'Darwin Kernel Version 13.4.0: Mon Jan 11 18:17:34 PST 2016; root:xnu-2422.115.15~1/RELEASE_X86_64', 'x86_64')2017-10-10 19:00:25,157 	pid=13505 	[assembly.py]	ERROR 	IndexError(list index out of range)


It stops in step 3:
 Step 3: Clustering/Mapping reads
  [####################] 100%  dereplicating         | 0:01:15    
  [####################] 100%  clustering            | 1:40:32  
  [####################] 100%  building clusters     | 0:00:25  
  [####################] 100%  chunking              | 0:00:03  
  [####################] 100%  aligning              | 7:25:21  

  Encountered an unexpected error (see ./ipyrad_log.txt)
  Error message is below -------------------------------
IndexError(list index out of range)


I have been running v0.7.13 successfully with a different dataset. I upgraded to 0.7.15 and am getting this error now, though with a different dataset. Everything I'm doing is denovo.

Thanks,
Angela

**Comment by isaacovercast** [comment]
Hi Angela,
This is definitely not a threading issue. Can you rerun step 3 and include the `-d` parameter to turn on debugging? After it crashes again can you post the last 20-30 lines of the ipyrad_log.txt file? 
-isaac

**Comment by isaacovercast** [comment]
Reopening the ticket and changing the title to reflect the current problem.

**Comment by angelajmcd** [comment]
Yep. It's running in debug now. Thanks!

**Comment by angelajmcd** [comment]
Hi Isaac,

It ran in debug, but now it looks like the error is different:

  loading Assembly: allsab1
  from saved path: ~/sabdata/allsab1.json
  establishing parallel connection:
  host compute node: [12 cores] on biol327-mac

  Step 1: Loading sorted fastq data to Samples
    Skipping: 285 Samples already found in Assembly allsab1.
    (can overwrite with force argument)    

  Step 2: Filtering reads 
    Skipping: All 285 selected Samples already edited.
    (can overwrite with force argument)    

  Step 3: Clustering/Mapping reads
  [####################] 100%  dereplicating         | 0:26:06    
  [####################] 100%  clustering            | 6:26:01  
  [####################] 100%  building clusters     | 0:00:34  

  Encountered an unexpected error (see ./ipyrad_log.txt)
  Error message is below -------------------------------
ValueError(too many values to unpack)



and here is the log.txt file for just that run: 


ipyrad [v.0.7.15]
  Interactive assembly and analysis of RAD-seq data
 -------------------------------------------------------------
  Begin run: 2017-10-11 10:55
  Using args {'preview': False, 'force': False, 'threads': 2, 'results': False, 'quiet': False, 'merge': None, 'ipcluster': None, 'cores': 0, 'params': 'params-allsab1.txt', 'branch': None, 'steps': '1234567', 'debug': False, 'new': None, 'MPI': False}
  Platform info: ('Darwin', 'biol327-mac', '13.4.0', 'Darwin Kernel Version 13.4.0: Mon Jan 11 18:17:34 PST 2016; root:xnu-2422.115.15~1/RELEASE_X86_64', 'x86_64')

**Comment by isaacovercast** [comment]
Hi Angela,

The debug flag was not enabled for this run, as you can see from the log
file: "Using args {'preview': False, 'force': False, 'threads': 2,
'results': False, 'quiet': False, 'merge': None, 'ipcluster': None,
'cores': 0, 'params': 'params-allsab1.txt', 'branch': None, 'steps':
'1234567', __'debug': False__, 'new': None, 'MPI': False}". Can you run it
again and include the `-d` flag. The log file should have _lots_ of stuff
in it if the debug flag is turned on.

-isaac

**Comment by angelajmcd** [comment]
Hmm... the -d was definitely included (here's the full terminal window)... but I will try running it again with -d again.

biol327-mac:sabdata PostDoc$ ipyrad -p params-allsab1.txt -s 1234567 -d

  ** Enabling debug mode ** 

 -------------------------------------------------------------
  ipyrad [v.0.7.15]
  Interactive assembly and analysis of RAD-seq data
 -------------------------------------------------------------
  loading Assembly: allsab1
  from saved path: ~/sabdata/allsab1.json
  establishing parallel connection:
  host compute node: [12 cores] on biol327-mac

  Step 1: Loading sorted fastq data to Samples
    Skipping: 285 Samples already found in Assembly allsab1.
    (can overwrite with force argument)    

  Step 2: Filtering reads 
    Skipping: All 285 selected Samples already edited.
    (can overwrite with force argument)    

  Step 3: Clustering/Mapping reads
  [####################] 100%  dereplicating         | 0:26:06    
  [####################] 100%  clustering            | 6:26:01  
  [####################] 100%  building clusters     | 0:00:34  

  Encountered an unexpected error (see ./ipyrad_log.txt)
  Error message is below -------------------------------
ValueError(too many values to unpack)

**Comment by angelajmcd** [comment]
Isaac,

Debug worked this time. Here is my terminal window followed by the log file:

biol327-mac:sabdata PostDoc$ ipyrad -p params-allsab1.txt -d -s 1234567 -f

  ** Enabling debug mode ** 

 -------------------------------------------------------------
  ipyrad [v.0.7.15]
  Interactive assembly and analysis of RAD-seq data
 -------------------------------------------------------------
  New Assembly: allsab1
  establishing parallel connection:
  host compute node: [12 cores] on biol327-mac

  Step 1: Loading sorted fastq data to Samples
  [####################] 100%  loading reads         | 0:09:49  
  285 fastq files loaded to 285 Samples.

  Step 2: Filtering reads 
  [####################] 100%  processing reads      | 2:21:41  

  Step 3: Clustering/Mapping reads
  [####################] 100%  dereplicating         | 0:00:02    
  [####################] 100%  clustering            | 1:40:25  
  [####################] 100%  building clusters     | 0:00:24  
  [####################] 100%  chunking              | 0:00:03  
  [####################] 100%  aligning              | 7:33:06  

  Encountered an unexpected error (see ./ipyrad_log.txt)
  Error message is below -------------------------------
IndexError(list index out of range)











 -------------------------------------------------------------
  ipyrad [v.0.7.15]
  Interactive assembly and analysis of RAD-seq data
 -------------------------------------------------------------
  Begin run: 2017-10-12 14:16
  Using args {'preview': False, 'force': False, 'threads': 2, 'results': False, 'quiet': False, 'merge': None, 'ipcluster': None, 'cores': 0, 'params': 'params-allsab1.txt', 'branch': None, 'steps': '1234567', 'debug': True, 'new': None, 'MPI': False}
  Platform info: ('Darwin', 'biol327-mac', '13.4.0', 'Darwin Kernel Version 13.4.0: Mon Jan 11 18:17:34 PST 2016; root:xnu-2422.115.15~1/RELEASE_X86_64', 'x86_64')2017-10-12 14:16:42,209 	pid=6628 	[load.py]	DEBUG 	skipping: no svd results present in old assembly
2017-10-12 14:16:43,352 	pid=6628 	[parallel.py]	INFO 	['ipcluster', 'start', '--daemonize', '--cluster-id=ipyrad-cli-6628', '--engines=Local', '--profile=default', '--n=12']
2017-10-12 14:16:59,845 	pid=6650 	[cluster_within.py]	INFO 	INSIDE derep ebu_BHY_6384
2017-10-12 14:16:59,846 	pid=6650 	[cluster_within.py]	INFO 	derep cmd ['/Volumes/Thunderbolt/Users/PostDoc/miniconda2/lib/python2.7/site-packages/bin/vsearch-osx-x86_64', '--derep_fulllength', '-', '--strand', 'plus', '--output', '/Volumes/Thunderbolt/Users/PostDoc/sabdata/allsab1_edits/ebu_BHY_6384_derep.fastq', '--threads', '2', '--fasta_width', '0', '--fastq_qmax', '1000', '--sizeout', '--relabel_md5']
2017-10-12 14:16:59,885 	pid=6650 	[cluster_within.py]	ERROR 	error inside derep_and_sort vsearch v2.0.5_osx_x86_64, 64.0GB RAM, 12 cores
https://github.com/torognes/vsearch



Fatal error: Unable to open output file for writing

2017-10-12 14:17:00,376 	pid=6658 	[cluster_within.py]	INFO 	INSIDE derep ebu_BHY_6387
2017-10-12 14:17:00,377 	pid=6658 	[cluster_within.py]	INFO 	derep cmd ['/Volumes/Thunderbolt/Users/PostDoc/miniconda2/lib/python2.7/site-packages/bin/vsearch-osx-x86_64', '--derep_fulllength', '-', '--strand', 'plus', '--output', '/Volumes/Thunderbolt/Users/PostDoc/sabdata/allsab1_edits/ebu_BHY_6387_derep.fastq', '--threads', '2', '--fasta_width', '0', '--fastq_qmax', '1000', '--sizeout', '--relabel_md5']
2017-10-12 14:17:00,389 	pid=6658 	[cluster_within.py]	ERROR 	error inside derep_and_sort vsearch v2.0.5_osx_x86_64, 64.0GB RAM, 12 cores
https://github.com/torognes/vsearch



Fatal error: Unable to open output file for writing

2017-10-12 14:17:00,764 	pid=6668 	[cluster_within.py]	INFO 	INSIDE derep chi_BHY_6441
2017-10-12 14:17:00,765 	pid=6668 	[cluster_within.py]	INFO 	derep cmd ['/Volumes/Thunderbolt/Users/PostDoc/miniconda2/lib/python2.7/site-packages/bin/vsearch-osx-x86_64', '--derep_fulllength', '-', '--strand', 'plus', '--output', '/Volumes/Thunderbolt/Users/PostDoc/sabdata/allsab1_edits/chi_BHY_6441_derep.fastq', '--threads', '2', '--fasta_width', '0', '--fastq_qmax', '1000', '--sizeout', '--relabel_md5']
2017-10-12 14:17:00,776 	pid=6668 	[cluster_within.py]	ERROR 	error inside derep_and_sort vsearch v2.0.5_osx_x86_64, 64.0GB RAM, 12 cores
https://github.com/torognes/vsearch



Fatal error: Unable to open output file for writing

2017-10-12 14:17:01,019 	pid=6660 	[cluster_within.py]	INFO 	INSIDE derep sej_GPO_6479
2017-10-12 14:17:01,020 	pid=6660 	[cluster_within.py]	INFO 	derep cmd ['/Volumes/Thunderbolt/Users/PostDoc/miniconda2/lib/python2.7/site-packages/bin/vsearch-osx-x86_64', '--derep_fulllength', '-', '--strand', 'plus', '--output', '/Volumes/Thunderbolt/Users/PostDoc/sabdata/allsab1_edits/sej_GPO_6479_derep.fastq', '--threads', '2', '--fasta_width', '0', '--fastq_qmax', '1000', '--sizeout', '--relabel_md5']
2017-10-12 14:17:01,032 	pid=6660 	[cluster_within.py]	ERROR 	error inside derep_and_sort vsearch v2.0.5_osx_x86_64, 64.0GB RAM, 12 cores
https://github.com/torognes/vsearch



Fatal error: Unable to open output file for writing

2017-10-12 14:17:01,732 	pid=6649 	[cluster_within.py]	INFO 	INSIDE derep chi_TAY_6223
2017-10-12 14:17:01,732 	pid=6649 	[cluster_within.py]	INFO 	derep cmd ['/Volumes/Thunderbolt/Users/PostDoc/miniconda2/lib/python2.7/site-packages/bin/vsearch-osx-x86_64', '--derep_fulllength', '-', '--strand', 'plus', '--output', '/Volumes/Thunderbolt/Users/PostDoc/sabdata/allsab1_edits/chi_TAY_6223_derep.fastq', '--threads', '2', '--fasta_width', '0', '--fastq_qmax', '1000', '--sizeout', '--relabel_md5']
2017-10-12 14:17:01,744 	pid=6649 	[cluster_within.py]	ERROR 	error inside derep_and_sort vsearch v2.0.5_osx_x86_64, 64.0GB RAM, 12 cores
https://github.com/torognes/vsearch



Fatal error: Unable to open output file for writing

2017-10-12 14:17:02,627 	pid=6658 	[cluster_within.py]	INFO 	INSIDE derep ind_BHY_6270
2017-10-12 14:17:02,627 	pid=6658 	[cluster_within.py]	INFO 	derep cmd ['/Volumes/Thunderbolt/Users/PostDoc/miniconda2/lib/python2.7/site-packages/bin/vsearch-osx-x86_64', '--derep_fulllength', '-', '--strand', 'plus', '--output', '/Volumes/Thunderbolt/Users/PostDoc/sabdata/allsab1_edits/ind_BHY_6270_derep.fastq', '--threads', '2', '--fasta_width', '0', '--fastq_qmax', '1000', '--sizeout', '--relabel_md5']
2017-10-12 14:17:02,633 	pid=6650 	[cluster_within.py]	INFO 	INSIDE derep wat_JAG_6308
2017-10-12 14:17:02,633 	pid=6650 	[cluster_within.py]	INFO 	derep cmd ['/Volumes/Thunderbolt/Users/PostDoc/miniconda2/lib/python2.7/site-packages/bin/vsearch-osx-x86_64', '--derep_fulllength', '-', '--strand', 'plus', '--output', '/Volumes/Thunderbolt/Users/PostDoc/sabdata/allsab1_edits/wat_JAG_6308_derep.fastq', '--threads', '2', '--fasta_width', '0', '--fastq_qmax', '1000', '--sizeout', '--relabel_md5']
2017-10-12 14:17:02,640 	pid=6658 	[cluster_within.py]	ERROR 	error inside derep_and_sort vsearch v2.0.5_osx_x86_64, 64.0GB RAM, 12 cores
https://github.com/torognes/vsearch



Fatal error: Unable to open output file for writing

2017-10-12 14:17:02,645 	pid=6650 	[cluster_within.py]	ERROR 	error inside derep_and_sort vsearch v2.0.5_osx_x86_64, 64.0GB RAM, 12 cores
https://github.com/torognes/vsearch



Fatal error: Unable to open output file for writing

2017-10-12 14:17:03,113 	pid=6656 	[cluster_within.py]	INFO 	INSIDE derep ebu_EBR_6350
2017-10-12 14:17:03,113 	pid=6656 	[cluster_within.py]	INFO 	derep cmd ['/Volumes/Thunderbolt/Users/PostDoc/miniconda2/lib/python2.7/site-packages/bin/vsearch-osx-x86_64', '--derep_fulllength', '-', '--strand', 'plus', '--output', '/Volumes/Thunderbolt/Users/PostDoc/sabdata/allsab1_edits/ebu_EBR_6350_derep.fastq', '--threads', '2', '--fasta_width', '0', '--fastq_qmax', '1000', '--sizeout', '--relabel_md5']
2017-10-12 14:17:03,125 	pid=6656 	[cluster_within.py]	ERROR 	error inside derep_and_sort vsearch v2.0.5_osx_x86_64, 64.0GB RAM, 12 cores
https://github.com/torognes/vsearch



Fatal error: Unable to open output file for writing

2017-10-12 14:17:04,605 	pid=6668 	[cluster_within.py]	INFO 	INSIDE derep ind_BHY_6275
2017-10-12 14:17:04,605 	pid=6668 	[cluster_within.py]	INFO 	derep cmd ['/Volumes/Thunderbolt/Users/PostDoc/miniconda2/lib/python2.7/site-packages/bin/vsearch-osx-x86_64', '--derep_fulllength', '-', '--strand', 'plus', '--output', '/Volumes/Thunderbolt/Users/PostDoc/sabdata/allsab1_edits/ind_BHY_6275_derep.fastq', '--threads', '2', '--fasta_width', '0', '--fastq_qmax', '1000', '--sizeout', '--relabel_md5']
2017-10-12 14:17:04,618 	pid=6668 	[cluster_within.py]	ERROR 	error inside derep_and_sort vsearch v2.0.5_osx_x86_64, 64.0GB RAM, 12 cores
https://github.com/torognes/vsearch



Fatal error: Unable to open output file for writing

2017-10-12 14:17:05,060 	pid=6649 	[cluster_within.py]	INFO 	INSIDE derep mel_BBG_6035
2017-10-12 14:17:05,061 	pid=6649 	[cluster_within.py]	INFO 	derep cmd ['/Volumes/Thunderbolt/Users/PostDoc/miniconda2/lib/python2.7/site-packages/bin/vsearch-osx-x86_64', '--derep_fulllength', '-', '--strand', 'plus', '--output', '/Volumes/Thunderbolt/Users/PostDoc/sabdata/allsab1_edits/mel_BBG_6035_derep.fastq', '--threads', '2', '--fasta_width', '0', '--fastq_qmax', '1000', '--sizeout', '--relabel_md5']
2017-10-12 14:17:05,061 	pid=6660 	[cluster_within.py]	INFO 	INSIDE derep mel_BBG_6036
2017-10-12 14:17:05,062 	pid=6660 	[cluster_within.py]	INFO 	derep cmd ['/Volumes/Thunderbolt/Users/PostDoc/miniconda2/lib/python2.7/site-packages/bin/vsearch-osx-x86_64', '--derep_fulllength', '-', '--strand', 'plus', '--output', '/Volumes/Thunderbolt/Users/PostDoc/sabdata/allsab1_edits/mel_BBG_6036_derep.fastq', '--threads', '2', '--fasta_width', '0', '--fastq_qmax', '1000', '--sizeout', '--relabel_md5']
2017-10-12 14:17:05,074 	pid=6649 	[cluster_within.py]	ERROR 	error inside derep_and_sort vsearch v2.0.5_osx_x86_64, 64.0GB RAM, 12 cores
https://github.com/torognes/vsearch



Fatal error: Unable to open output file for writing

2017-10-12 14:17:05,076 	pid=6660 	[cluster_within.py]	ERROR 	error inside derep_and_sort vsearch v2.0.5_osx_x86_64, 64.0GB RAM, 12 cores
https://github.com/torognes/vsearch



Fatal error: Unable to open output file for writing

2017-10-12 14:17:05,626 	pid=6656 	[cluster_within.py]	INFO 	INSIDE derep mel_BBG_6032
2017-10-12 14:17:05,626 	pid=6656 	[cluster_within.py]	INFO 	derep cmd ['/Volumes/Thunderbolt/Users/PostDoc/miniconda2/lib/python2.7/site-packages/bin/vsearch-osx-x86_64', '--derep_fulllength', '-', '--strand', 'plus', '--output', '/Volumes/Thunderbolt/Users/PostDoc/sabdata/allsab1_edits/mel_BBG_6032_derep.fastq', '--threads', '2', '--fasta_width', '0', '--fastq_qmax', '1000', '--sizeout', '--relabel_md5']
2017-10-12 14:17:05,638 	pid=6656 	[cluster_within.py]	ERROR 	error inside derep_and_sort vsearch v2.0.5_osx_x86_64, 64.0GB RAM, 12 cores
https://github.com/torognes/vsearch



Fatal error: Unable to open output file for writing

2017-10-12 14:17:06,214 	pid=6650 	[cluster_within.py]	INFO 	INSIDE derep mel_MCA_6096
2017-10-12 14:17:06,215 	pid=6650 	[cluster_within.py]	INFO 	derep cmd ['/Volumes/Thunderbolt/Users/PostDoc/miniconda2/lib/python2.7/site-packages/bin/vsearch-osx-x86_64', '--derep_fulllength', '-', '--strand', 'plus', '--output', '/Volumes/Thunderbolt/Users/PostDoc/sabdata/allsab1_edits/mel_MCA_6096_derep.fastq', '--threads', '2', '--fasta_width', '0', '--fastq_qmax', '1000', '--sizeout', '--relabel_md5']
2017-10-12 14:17:06,225 	pid=6650 	[cluster_within.py]	ERROR 	error inside derep_and_sort vsearch v2.0.5_osx_x86_64, 64.0GB RAM, 12 cores
https://github.com/torognes/vsearch

**Comment by 0seastar0** [comment]
I am getting the same error on step 3. Following.

**Comment by isaacovercast** [comment]
`Fatal error: Unable to open output file for writing`

This is almost certainly a disk space issue. You have 285 samples, and step 3 creates a lot of intermediate files, so this is going to consume a TON of disk space. Verify that you have sufficient disk space and then rerun it.

**Comment by TomaszSuchan** [comment]
Hi, I'm having a similar issue with ipyrad 0.7.15.

The end of the logfile looks like that:

```
2017-10-17 17:08:58,299         pid=14564       [cluster_within.py]     INFO    chunk ['/home/tsuchan/Projects/radz/radz-tmpalign/A_nuda_AVE671_B7_chunk_0.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_nuda_AVE671_B7_chunk_1.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_nuda_AVE671_B7_chunk_2.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_nuda_AVE671_B7_chunk_3.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_nuda_AVE671_B7_chunk_4.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_nuda_AVE671_B7_chunk_5.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_nuda_AVE671_B7_chunk_6.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_nuda_AVE671_B7_chunk_7.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_nuda_AVE671_B7_chunk_8.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_nuda_AVE671_B7_chunk_9.aligned']
2017-10-17 17:09:40,076         pid=14597       [cluster_within.py]     INFO    chunk ['/home/tsuchan/Projects/radz/radz-tmpalign/A_hirtula_52150_chunk_0.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_hirtula_52150_chunk_1.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_hirtula_52150_chunk_2.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_hirtula_52150_chunk_3.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_hirtula_52150_chunk_4.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_hirtula_52150_chunk_5.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_hirtula_52150_chunk_6.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_hirtula_52150_chunk_7.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_hirtula_52150_chunk_8.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_hirtula_52150_chunk_9.aligned']
2017-10-17 17:10:15,709         pid=14595       [cluster_within.py]     INFO    chunk ['/home/tsuchan/Projects/radz/radz-tmpalign/A_fatua_52335_chunk_0.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_fatua_52335_chunk_1.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_fatua_52335_chunk_2.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_fatua_52335_chunk_3.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_fatua_52335_chunk_4.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_fatua_52335_chunk_5.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_fatua_52335_chunk_6.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_fatua_52335_chunk_7.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_fatua_52335_chunk_8.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_fatua_52335_chunk_9.aligned']
2017-10-17 17:12:58,121         pid=14589       [cluster_within.py]     INFO    chunk ['/home/tsuchan/Projects/radz/radz-tmpalign/A_insularis_52439_chunk_0.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_insularis_52439_chunk_1.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_insularis_52439_chunk_2.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_insularis_52439_chunk_3.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_insularis_52439_chunk_4.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_insularis_52439_chunk_5.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_insularis_52439_chunk_6.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_insularis_52439_chunk_7.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_insularis_52439_chunk_8.aligned', '/home/tsuchan/Projects/radz/radz-tmpalign/A_insularis_52439_chunk_9.aligned']
2017-10-17 17:12:58,195         pid=14376       [assembly.py]   ERROR   IndexError(list index out of range)
2017-10-17 17:12:59,260         pid=14376       [assembly.py]   INFO    interrupted engine 4 w/ SIGINT to 14589
2017-10-17 17:13:00,277         pid=14376       [assembly.py]   INFO      shutting down engines
2017-10-17 17:13:02,022         pid=14376       [assembly.py]   INFO      finished shutdown
2017-10-17 17:13:02,029         pid=14376       [__init__.py]   INFO    debugging turned off
```

**Comment by angelajmcd** [comment]
Hi again,

I was running it on a disk that had 500 gb free previously. I restarted it overnight on a disk with 4 TB free, and got the same error... I'm not sure that this is a disk space issue.  I will restart it again in debug and see what it says again.

**Comment by isaacovercast** [comment]
Ok, i'm starting to think this isn't a disk issue. Can one or the other of you please dropbox me your raw data and a params file so i can debug this? Or wetransfer, whatever works.

**Comment by angelajmcd** [comment]
Sure. I'll send it via dropbox to iovercast@gc.cuny.edu

**Comment by isaacovercast** [comment]
Great!

On Wed, Oct 18, 2017 at 10:57 AM, Angela McDonnell <notifications@github.com

**Comment by 0seastar0** [comment]
I sent my data as well. Hope it helps. Thanks!

**Comment by isaacovercast** [comment]
The issue associated with this ticket was fixed, I just didn't close it. If you are seeing this same error can you post more information about error message, data type, contents of the log file, and anything else you think would be helpful?

**Comment by tuckerwc91** [comment]
The error is still occurring after step 3. It comes us as "Caught error while decloning 3rad data - list index out of range". I'm running already de-multiplexed paired files. My log file doesn't seem to have the same error messages that some of the other users had with this error. I've attached my log file. 

[ipyrad_log.txt](https://github.com/dereneaton/ipyrad/files/1911599/ipyrad_log.txt)

**Comment by tuckerwc91** [comment]
I'm also running this assembly denovo

**Comment by isaacovercast** [comment]
Ok. Well this is a different problem from the one mentioned in this ticket so i created a new issue for it: #291

**Comment by isaacovercast** [comment]
I assume this actually _is_ 3Rad data, yes? Just doublechecking because 3rad is a little unusual.

**Comment by amesclir** [comment]
The same problem o similar here:

####################
####################

loading Assembly: carexfurva
  from saved path: ~/floragenex2020/carexfurva.json

 -------------------------------------------------------------
  ipyrad [v.0.9.65]
  Interactive assembly and analysis of RAD-seq data
 -------------------------------------------------------------
  Parallel connection | bigfoot: 16 cores

  Step 3: Clustering/Mapping reads within samples
^M  [                    ]   0% 0:00:00 | indexing reference   ^M  [           $
^M  [##################  ]  93% 0:00:16 | dereplicating        ^M  [###########$

  Encountered an Error.
  Message: IndexError: list index out of range
  Parallel connection closed.
  loading Assembly: carexfurva
  from saved path: ~/floragenex2020/carexfurva.json

##############################
###############################

This is an older version 0.9.65

This is supposed to be fixed, right?

**Comment by amesclir** [comment]
Here is with the -d flag

#######################
#######################


loading Assembly: mergecarexfl
  from saved path: ~/floragenex2020/mergecarexfl.json

 -------------------------------------------------------------
  ipyrad [v.0.9.65]
  Interactive assembly and analysis of RAD-seq data
 -------------------------------------------------------------
  Parallel connection | nodo93: 16 cores

  Step 3: Clustering/Mapping reads within samples
^M  [                    ]   0% 0:00:00 | indexing reference   ^M  [                    ]   0% 0:00:00 | indexing reference   ^M  [                    ]   0% 0:00:00 | indexing reference   ^M  [	  $
^M  [################### ]  97% 0:00:13 | concatenating        ^M  [################### ]  97% 0:00:13 | concatenating        ^M  [################### ]  98% 0:00:13 | concatenating        ^M  [########$
^M  [##################  ]  94% 0:00:16 | dereplicating        ^M  [################### ]  96% 0:00:16 | dereplicating        ^M  [################### ]  96% 0:00:16 | dereplicating        ^M  [########$

  Encountered an Error.
  Message: IndexError: list index out of range
  Parallel connection closed.
^[[0;31m---------------------------------------------------------------------------^[[0m^[[0;31mIndexError^[[0m                                Traceback (most recent call last)^[[0;32m<string>^[[0m in ^$
^[[0;32m/home/software/anaconda3/envs/ipyrad-0.9.65/lib/python3.8/site-packages/ipyrad/assemble/clustmap.py^[[0m in ^[[0;36mdereplicate^[[0;34m(data, sample, nthreads)^[[0m
^[[1;32m    737^[[0m     ]
^[[1;32m    738^[[0m     ^[[0minfiles^[[0m ^[[0;34m=^[[0m ^[[0;34m[^[[0m^[[0mi^[[0m ^[[0;32mfor^[[0m ^[[0mi^[[0m ^[[0;32min^[[0m ^[[0minfiles^[[0m ^[[0;32mif^[[0m ^[[0mos^[[0m^[[0;34m.^[[0m^[[0mpath^[[0$
^[[0;32m--> 739^[[0;31m     ^[[0minfile^[[0m ^[[0;34m=^[[0m ^[[0minfiles^[[0m^[[0;34m[^[[0m^[[0;34m-^[[0m^[[0;36m1^[[0m^[[0;34m]^[[0m^[[0;34m^[[0m^[[0;34m^[[0m^[[0m
^[[0m^[[1;32m    740^[[0m ^[[0;34m^[[0m^[[0m
^[[1;32m    741^[[0m     ^[[0;31m# datatypes options^[[0m^[[0;34m^[[0m^[[0;34m^[[0m^[[0;34m^[[0m^[[0m
^[[0;31mIndexError^[[0m: list index out of range

**Comment by isaacovercast** [comment]
@amesclir Did you check that you have sufficient disk space? Running out of disk will cause this problem. I would also recommend updating to the newest version of ipyrad `0.9.88` and trying again, but also yes, check your disk space.

**Comment by isaacovercast** [comment]
Is it happening at the same place every time? It looks like a rather small dataset (it runs really really fast it seems), if you want to share your raw data with me I can take a look (wetransfer). Is it an option to try running it on a different machine?

**Comment by amesclir** [comment]
Dear Isaac,

We are running in a cluster from my university. It would be not very
practical for us to try on another computer.
In this case, we run step 1 and 2 in two plates from two different lanes,
then we merge, and the error cans in step 3 after merging (we use a
reference genome).
The raw data is large, like 70 gb or something like that.
Maybe I could also sent you the edits folders? And you have a look on
there?
(I would not mind to give you the username and password and you access
directly to cluster remotely)
Please, let me know.
Thanks!
Marcial

**Comment by avalflo** [comment]
Hello, I’m using ipyrad version 0.9.88 and I get the same error in step3 (here with -d):

  Encountered an Error.
  Message: IndexError: list index out of range
  Parallel connection closed.
^[[0;31m---------------------------------------------------------------------------^[[0m
^[[0;31mIndexError^[[0m                                Traceback (most recent call last)
File ^[[0;32m<string>:1^[[0m

File ^[[0;32m/home/software/anaconda3/envs/ipyrad-0.9.88/lib/python3.10/site-packages/ipyrad/assemble/clustmap.py:741^[[0m, in ^[[0;36mdereplicate^[[0;34m(data, sample, nthreads)^[[0m
^[[1;32m    726^[[0m infiles ^[[38;5;241m=^[[39m [
^[[1;32m    727^[[0m     os^[[38;5;241m.^[[39mpath^[[38;5;241m.^[[39mjoin(
^[[1;32m    728^[[0m         data^[[38;5;241m.^[[39mdirs^[[38;5;241m.^[[39medits,
^[[0;32m   (...)^[[0m
^[[1;32m    738^[[0m         ^[[38;5;124m"^[[39m^[[38;5;132;01m{}^[[39;00m^[[38;5;124m_declone.fastq^[[39m^[[38;5;124m"^[[39m^[[38;5;241m.^[[39mformat(sample^[[38;5;241m.^[[39mname)),
^[[1;32m    739^[[0m ]
^[[1;32m    740^[[0m infiles ^[[38;5;241m=^[[39m [i ^[[38;5;28;01mfor^[[39;00m i ^[[38;5;129;01min^[[39;00m infiles ^[[38;5;28;01mif^[[39;00m os^[[38;5;241m.^[[39mpath^[[38;5;241m.^[[39mexists(i)]
^[[0;32m--> 741^[[0m infile ^[[38;5;241m=^[[39m ^[[43minfiles^[[49m^[[43m[^[[49m^[[38;5;241;43m-^[[39;49m^[[38;5;241;43m1^[[39;49m^[[43m]^[[49m
^[[1;32m    743^[[0m ^[[38;5;66;03m# datatypes options^[[39;00m
^[[1;32m    744^[[0m strand ^[[38;5;241m=^[[39m ^[[38;5;124m"^[[39m^[[38;5;124mplus^[[39m^[[38;5;124m"^[[39m

^[[0;31mIndexError^[[0m: list index out of range

I’m executing the analyses in an HPC cluster, with a -pe smp 16. I have already checked the disk space and it doesn’t seem to be the problem. I tried also with oldest versions, but I always get an error… 

How can I solve this?

Thank you!

**Comment by isaacovercast** [comment]
@avalflo Can you please try updating to the most recent version of ipyrad (0.9.92) and then try again? I fixed a similar issue a few months ago (#502), so your issue here might already be fixed in the newest version.

**Comment by avalflo** [comment]
Hello, 
I tried again with version 0.9.92 and I got this error (with -d): 

  Step 3: Clustering/Mapping reads within samples
^M  [                    ]   0% 0:00:00 | indexing reference   ^M  [                    ]   0% 0:00:00 | indexing reference   ^M  [                    ]   0$
^M  [################### ]  97% 0:00:13 | concatenating        ^M  [################### ]  98% 0:00:13 | concatenating        ^M  [################### ]  99$
^M  [#                   ]   7% 0:00:15 | dereplicating        ^M  [#                   ]   8% 0:00:15 | dereplicating        ^M  [#                   ]   8$
^M  [                    ]   0% 0:00:13 | mapping reads        ^M  [                    ]   0% 0:00:13 | mapping reads        ^M  [                    ]   0$

  Encountered an Error.
  Message: IPyradError: bwa error: None
  Parallel connection closed.
^[[0;31m---------------------------------------------------------------------------^[[0m
^[[0;31mIPyradError^[[0m                               Traceback (most recent call last)
File ^[[0;32m<string>:1^[[0m

File ^[[0;32m/home/software/anaconda3/envs/ipyrad-0.9.92.2/lib/python3.10/site-packages/ipyrad/assemble/clustmap.py:1935^[[0m, in ^[[0;36mmapping_reads^[[0;$
^[[1;32m   1933^[[0m     error1 ^[[38;5;241m=^[[39m proc1^[[38;5;241m.^[[39mcommunicate()[^[[38;5;241m0^[[39m]
^[[1;32m   1934^[[0m     ^[[38;5;28;01mif^[[39;00m proc1^[[38;5;241m.^[[39mreturncode:
^[[0;32m-> 1935^[[0m         ^[[38;5;28;01mraise^[[39;00m IPyradError(^[[38;5;124m"^[[39m^[[38;5;124mbwa error: ^[[39m^[[38;5;132;01m{}^[[39;00m^[[38;5;124m$
^[[1;32m   1937^[[0m ^[[38;5;66;03m# sends unmapped reads to a files and will PIPE mapped reads to cmd3^[[39;00m
^[[1;32m   1938^[[0m cmd2 ^[[38;5;241m=^[[39m [
^[[1;32m   1939^[[0m     ip^[[38;5;241m.^[[39mbins^[[38;5;241m.^[[39msamtools, ^[[38;5;124m"^[[39m^[[38;5;124mview^[[39m^[[38;5;124m"^[[39m,
^[[1;32m   1940^[[0m     ^[[38;5;124m"^[[39m^[[38;5;124m-b^[[39m^[[38;5;124m"^[[39m,
^[[0;32m   (...)^[[0m
^[[1;32m   1944^[[0m     samout,
^[[1;32m   1945^[[0m ]

^[[0;31mIPyradError^[[0m: bwa error: None

It seems that the error ocurrs in bwa package while mapping reads. I am using a reference genome, could that be the problem?

Thank you!
Ana

**Comment by isaacovercast** [comment]
What format is the reference sequence in? Where did it come from? If the reference sequence is not fasta or if it is unusual in any way then the indexing will fail. You can try running the bwa index by hand to see if there are errors:

`bwa index ref.fa`
